tensorrt_llm/llmapi/trtllm-llmapi-launch

```diff
@@ -23,17 +23,28 @@ function mpi_world_size {
     fi
 }
 
-function export_free_tcp_addr {
-    # find free port with python -c, the port should start from 10000
-    local free_port=$(python -c 'import socket; s=socket.socket(); s.bind(("", 10000)); print(s.getsockname()[1]); s.close()')
+function export_free_tcp_addr_for_spawn_proxy_process {
+    # find free port starting from 10012
+    local free_port=$(python -c 'import socket; s=socket.socket();
+port = 10012
+while True:
+    try:
+        s.bind(("", port))
+        break
+    except OSError:
+        port += 1
+print(port); s.close()')
     export TLLM_SPAWN_PROXY_PROCESS_IPC_ADDR="tcp://127.0.0.1:${free_port}"
     log_stderr "TLLM_SPAWN_PROXY_PROCESS_IPC_ADDR: $TLLM_SPAWN_PROXY_PROCESS_IPC_ADDR"
+
+    export TLLM_SPAWN_PROXY_PROCESS_IPC_HMAC_KEY=$(openssl rand -hex 32)
 }
 
+
 export tllm_mpi_size=$(mpi_world_size)
 log_stderr "tllm_mpi_size: $tllm_mpi_size"
 
-export_free_tcp_addr
+export_free_tcp_addr_for_spawn_proxy_process
```

Comment:
Nit @Superjomn  I think we have discussed this, it only beeds to be export in mpi rank 0?

Do you think its a safe change here? I am fine if you'd like to do it in another change.

Reply:
Yeah, I will perform this change if the current CI fails, or I will change it in a subsequent PR.